File: cpp-thin-client/src/banner.cpp
#include <iostream>
#include <fstream>
#include <string>

namespace ironwall {
void print_banner() {
    std::ifstream in("../assets/banner.txt");
    if (!in) in.open("../../assets/banner.txt");
    if (!in) in.open("assets/banner.txt");
    if (in) {
        std::string line;
        while (std::getline(in, line)) std::cout << line << "\n";
    } else {
        std::cout << "=== IRONWALL Anti-Cheat ===\n";
    }
    std::cout << std::endl;
}
}

File: rust-thin-client/src/banner.rs
pub fn print_banner() {
    let candidates = [
        "assets/banner.txt",
        "../assets/banner.txt",
        "../../assets/banner.txt",
    ];
    for path in candidates {
        if let Ok(s) = std::fs::read_to_string(path) {
            println!("{s}");
            return;
        }
    }
    println!("=== IRONWALL Anti-Cheat ===\n");
}
